whylog/config/investigation_plan.py

```diff
@@ -50,9 +50,26 @@ def compare_with_bound(self, bound, super_parser_groups):
         Basing on super_parser_groups extracted from line, returns information
         how relative to bound this line is.
         """
-        #TODO: remove mock
+        # This implementation assume that super_parser_groups length equals 1 or 0
```

Comment:
I think you should split this function into two ones - one that takes bound information (like now) and the second one that takes value and compare it with another value. The first function should extract the concrete values from InvestigationStep and the second one shouldn't even know if it is left or right bound.